flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/table/descriptors/Kafka.java

```diff
@@ -265,24 +259,24 @@ public Kafka sinkPartitionerCustom(Class<? extends FlinkKafkaPartitioner> partit
 		}
 
 		if (specificOffsets != null) {
-			final List<List<String>> values = new ArrayList<>();
+			final StringBuilder stringBuilder = new StringBuilder();
+			int i = 0;
 			for (Map.Entry<Integer, Long> specificOffset : specificOffsets.entrySet()) {
-				values.add(Arrays.asList(specificOffset.getKey().toString(), specificOffset.getValue().toString()));
+				if (i != 0) {
+					stringBuilder.append(';');
+				}
+				stringBuilder.append(CONNECTOR_SPECIFIC_OFFSETS_PARTITION).append(':').append(specificOffset.getKey()).append(',');
+				stringBuilder.append(CONNECTOR_SPECIFIC_OFFSETS_OFFSET).append(':').append(specificOffset.getValue());
+				i++;
 			}
-			properties.putIndexedFixedProperties(
-				CONNECTOR_SPECIFIC_OFFSETS,
-				Arrays.asList(CONNECTOR_SPECIFIC_OFFSETS_PARTITION, CONNECTOR_SPECIFIC_OFFSETS_OFFSET),
-				values);
+			properties.putString(CONNECTOR_SPECIFIC_OFFSETS, stringBuilder.toString());
 		}
 
 		if (kafkaProperties != null) {
-			properties.putIndexedFixedProperties(
-				CONNECTOR_PROPERTIES,
-				Arrays.asList(CONNECTOR_PROPERTIES_KEY, CONNECTOR_PROPERTIES_VALUE),
-				this.kafkaProperties.entrySet().stream()
-					.map(e -> Arrays.asList(e.getKey(), e.getValue()))
-					.collect(Collectors.toList())
-				);
+			this.kafkaProperties.entrySet()
+					.forEach(
```

Comment:
3 line can be 1 line?

Reply:
like this `this.kafkaProperties.entrySet().forEach(entry -> ` ?

Reply:
Yes, I think these lines can be folded.